src/Container/InMemoryEventStoreFactory.php

```diff
@@ -154,7 +170,44 @@ public function defaultOptions(): iterable
             'metadata_enrichers' => [],
             'plugins' => [],
             'wrap_action_event_emitter' => true,
+            'transactional' => true,
             'read_only' => false,
         ];
     }
+
+    private function determineEventsForDefaultEmitter(): array
+    {
+        if ($this->isTransactional) {
+            return TransactionalActionEventEmitterEventStore::ALL_EVENTS;
+        }
+
+        return ActionEventEmitterEventStore::ALL_EVENTS;
+    }
+
+    private function createEventStore(): EventStore
+    {
+        if ($this->isTransactional) {
+            return new InMemoryEventStore();
+        }
+
+        return new NonTransactionalInMemoryEventStore();
+    }
+
+    private function createActionEventEmitterDecorator(
+        EventStore $eventStore,
+        ActionEventEmitter $actionEventEmitter
+    ): ActionEventEmitterEventStore {
+        if ($this->isTransactional) {
+            assert($eventStore instanceof TransactionalEventStore);
```

Comment:
actually, the TransactionalActionEventEmitterEventStore expects TransactionalEventStore as argument in the constructor, so this can be removed as well